src/hotspot/share/memory/iterator.inline.hpp

```diff
@@ -51,12 +50,7 @@ inline void ClaimMetadataVisitingOopIterateClosure::do_cld(ClassLoaderData* cld)
 }
 
 inline void ClaimMetadataVisitingOopIterateClosure::do_klass(Klass* k) {
-  ClassLoaderData* cld = k->class_loader_data();
-  if (cld != nullptr) {
-    ClaimMetadataVisitingOopIterateClosure::do_cld(cld);
-  } else {
-    assert(AOTLinkedClassBulkLoader::is_pending_aot_linked_class(k), "sanity");
```

Comment:
Why is the null check and assert not valid anymore?

Reply:
The AOT cache can have heap objects whose class are loaded by the platform/app loaders. These objects usually come from aot-linked Lambda expressions.

Before this PR, the platform/app classes are loaded after executing quite a bit of Java code (i.e.,module graph initialization). It's possible for a GC to happen and discover those heap objects, whose classes are not yet loaded (i.e., having a null `cld`).

After this PR, all boot/platform/app classes are loaded before GC can happen, so we will never run into such null `cld` anymore.

Reply:
That makes `is_pending_aot_linked_class` dead code. Can it be removed?